core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1210,27 +1210,29 @@ public static function redirect($strLocation, $intStatus=303)
 	 */
 	protected static function replaceOldBePaths($strContext)
 	{
-		$router = System::getContainer()->get('router');
+		$arrMapper = array
+		(
+			'contao/confirm.php'   => 'contao_backend_confirm',
+			'contao/file.php'      => 'contao_backend_file',
+			'contao/help.php'      => 'contao_backend_help',
+			'contao/index.php'     => 'contao_backend_login',
+			'contao/main.php'      => 'contao_backend',
+			'contao/page.php'      => 'contao_backend_page',
+			'contao/password.php'  => 'contao_backend_password',
+			'contao/popup.php'     => 'contao_backend_popup',
+			'contao/preview.php'   => 'contao_backend_preview',
+		);
 
-		$generate = static function ($route) use ($router)
+		$replace = static function ($matches) use ($arrMapper)
 		{
-			return substr($router->generate($route), \strlen(Environment::get('path')) + 1);
+			$router = System::getContainer()->get('router');
```

Comment:
Just a nitpick: Wouldn‘t it be more performant if we retrieved the router service outside the callable?

Reply:
no, because then we would always retrieve it even if the callback is not called. The callback will never be called multiple times (you usually don't have more than one old entry point in one URL 😉 )

Reply:
Not exactly true, it happens for every single back end template :) But it still doesn't matter^^

Reply:
hmmm, I actually didn't think about that. That means we will generate the same URL multiple times if it appears multiple times. Guess I need to change the PR again 🙈